lib/Cake/Controller/Component/Auth/BasicAuthenticate.php

```diff
@@ -82,8 +82,17 @@ public function authenticate(CakeRequest $request, CakeResponse $response) {
  * @return mixed Either false or an array of user information
  */
 	public function getUser(CakeRequest $request) {
-		$username = env('PHP_AUTH_USER');
-		$pass = env('PHP_AUTH_PW');
+		$username = '';
+		$pass = '';
+		if (isset($_SERVER['PHP_AUTH_USER'])) {
```

Comment:
The efficiency here is irrelevant. Each adapter is invoked never in loops. 
It is better to stay consistent across the code base.